src/components/layout/footer/disclaimer.tsx

```diff
@@ -9,14 +9,14 @@ import {
 import { Mobile, Desktop } from 'components/containers'
 import { Localize, localize } from 'components/localization'
 import { loss_percent } from 'common/constants'
-import { useCountryRule } from 'components/hooks/use-country-rule'
+import { RegionContext } from 'store/region-context'
 
 type DisclaimerSectionProps = {
     is_academy?: boolean
 }
 
 const DisclaimerSection = ({ is_academy = false }: DisclaimerSectionProps) => {
-    const { is_eu, is_non_eu } = useCountryRule()
+    const { is_eu, is_non_eu } = React.useContext(RegionContext)
```

Comment:
@mohammad-rework Not sure what’s the reason behind this, But we create abstractions in the code to avoid these kinds of changes.
So we already had the `useCountryRule` abstraction and we could just refactor the logic inside without making changes in other places. Now we have +100 file changes which are only changing the `useCountryRule` with `React.useContext(RegionContext)` 🤔

Reply:
Also instead of exposing the `RegionContext` directly and using `useContext` everywhere, we should create an abstraction like `useRegion` and also do error handling for when the context is not provided.